ruby_event_store/lib/ruby_event_store/specification.rb

```diff
@@ -33,14 +33,36 @@ def from(start)
     # Limits the query to events before or after another event.
     # {http://railseventstore.org/docs/read/ Find out more}.
     #
-    # @param start [String] id of event to start reading from.
+    # @param stop [String] id of event to start reading from.
     # @return [Specification]
     def to(stop)
       raise InvalidPageStop if stop.nil? || stop.empty?
       raise EventNotFound.new(stop) unless reader.has_event?(stop)
       Specification.new(reader, result.dup { |r| r.stop = stop })
     end
 
+    # Limits the query to events before or after another event.
+    # {http://railseventstore.org/docs/read/ Find out more}.
+    #
+    # @param date [Date]
+    # @param equals [Boolean] whether you want to include events on that date
+    # @return [Specification]
+    def older_than(date, equals = false)
+      raise InvalidPageStart if date.nil? || date.empty?
+      Specification.new(reader, result.dup { |r| r.older_than = [date, equals] })
+    end
+
+    # Limits the query to events before or after another event.
+    # {http://railseventstore.org/docs/read/ Find out more}.
+    #
+    # @param date [Date]
+    # @param equals [Boolean] whether you want to include events on that date
+    # @return [Specification]
+    def newer_than(date, equals = false)
```

Comment:
`newer_than(date, equals = false)` - I don't like that 2nd argument - it adds additional complexity here, we need to store tuple `<Date, Boolean>` instead of just `Date` (or maybe `Time` as mentioned above), then `specification.newer_than` returns that tuple instead of just time. Let's decide on some way of selecting events (>& < as method names suggests) and let developers to pass appropriate time range.

Reply:
that's what I wanted input on. Not very happy with that API. Another option was to expose 4 methods to map the `<`, `<=`, `>` and `>=` operators. Also not sure about the method name. Maybe we could go with something like arel: `lt`, `lteq`, `gt`, `gteq`?

Reply:
I would keep the naming and let developer to provide valid arguments for `>` & `<` operators.

Reply:
WIth Ruby 2.7 endless ranges it could be interesting to consider `between(Range)` API.

Reply:
Yes, I though about it but we want to have open ranges and we still support Ruby 2.4 AFAIR :)